dali/python/nvidia/dali/pipeline.py

```diff
@@ -965,9 +965,9 @@ def schedule_run(self):
         Should not be mixed with :meth:`run` in the same pipeline"""
         with self._check_api_type_scope(types.PipelineAPIType.SCHEDULED):
             if self._first_iter and self._exec_pipelined:
-                self._prefetch()
+                self._prefetch(**pipeline_inputs)
```

Comment:
That's bad.... It would most likely cause very weird behavior of the pipeline and explaining it to the user would be extremely hard. Instead, rather enforce that queue depth is 1 when using pipeline inputs.

Reply:
Unfortunately, with how currently the running logic is organised, it has to be there. `if self._first_iter and self._exec_pipelined:` means, that when we legitimately set `exec_pipelined=True` and `prefetch_queue_depth=1`, the `prefetch` function will be executed. And we want to have `pipeline_inputs` in this scenario.

I've added the enforce and extensive explanation in the `run` function. Hopefully this will solve the issue.

Reply:
(to put our call in writing):
We don't have to propagate it down to `_prefetch` and `_run_once` - we could just call `feed_input` in `schedule_run` or even restrict the new API to run, so we avoid _yet another_ way to feed the (prefetching) pipeline.